src/Extension/FluentExtension.php

```diff
@@ -629,6 +629,46 @@ public function onAfterWrite(): void
     {
         $this->handleClassChanged();
     }
+    
+    /**
+     * If an object is duplicated also duplicate existing localised values from original to new object.
+     */
+    public function onAfterDuplicate($original, $doWrite, $relations): void
+    {
+        $localisedTables = $this->owner->getLocalisedTables();
+
+        $suffixes = [''];
+        if ($this->owner->hasExtension(FluentVersionedExtension::class)) {
+            $suffixes = ['', FluentVersionedExtension::SUFFIX_LIVE, FluentVersionedExtension::SUFFIX_VERSIONS];
+        }
+
+        foreach ($suffixes as $suffix) {
+            foreach ($localisedTables as $tableName => $fields) {
+                // Target IDs
+                $fromID = $original->ID;
+                $toID = $this->owner->ID;
+
+                // Get table and add suffix
+                $localisedTable = $this->getLocalisedTable($tableName);
+                $localisedTable .= $suffix;
+
+                // Also copy Version field if table is versions table
+                if ($suffix === FluentVersionedExtension::SUFFIX_VERSIONS) {
+                    $fields[] = 'Version';
+                }
+
+                // Remove existing translations from duplicated object, created by onBeforeWrite
+                DB::query("DELETE FROM $localisedTable WHERE RecordID=$toID");
```

Comment:
can you please parametise all arguments, and ensure all identifiers are ansi sql quoted (double quotes).

```
DB::prepared_query("DELETE FROM \"$localisedTable\" WHERE \"RecordID\" = ?", [$toID]);
```

Reply:
Won't this query fail if $suffix == "", since there is no RecordID column on tables other than _Versions suffixed one?